Sources/ClientRuntime/Telemetry/Logging/SDKLoggingSystem.swift

```diff
@@ -7,20 +7,34 @@
 
 import Logging
 
+/// Use this to turn SDK logging on.
 public actor SDKLoggingSystem {
     private var isInitialized = false
-    private var factories: [String: SDKLogHandlerFactory] = [:]
+    private var logHandlerFactories: [String: SDKLogHandlerFactory] = [:]
 
     public init() {}
 
+    /// Adds custom log handler factory to `this.logHandlerFactories`.
+    ///
+    /// The added log handler factory will be dedicated log handler for any logger with identical label.
     public func add(logHandlerFactory: SDKLogHandlerFactory) {
-        let label = logHandlerFactory.label
-        factories[label] = logHandlerFactory
+        logHandlerFactories[logHandlerFactory.label] = logHandlerFactory
     }
 
+    /// Initializes the logging handler factory for the SDK.
+    /// The default behavior is to log messages at `.error` or more severe levels.
+    ///
+    /// The handler factory closure first checks if there exists a custom handler factory in `this.logHandlerFactories` with the same label as the label given to logger initializer. If it exists, then that factory gets used to create the log handler.
+    ///
+    /// If no custom handler factory is found for the given label, the factory closure creates and returns a `StreamLogHandler` with minimum log level set to `logLevel`.
+    ///
+    /// Loggers output log only if the log level of the message is equal to or more severe than the underlying log handler's log level. E.g., `logger.info(...)` executes only if the underlying log handler's log level is `.info`, `.debug`, or `.trace`. It does not execute if the underlying log handler's minimum log level is any one of the following levels that are more severe than `.info`: `.notice`, `.warning`, `error`, `critical`.
+    ///
+    /// - parameters:
+    ///     - logLevel: The minimum log level to use for the log handler if no custom log handler factory was found. Default is `.error`.
     public func initialize(defaultLogLevel: SDKLogLevel = .error) async {
         if isInitialized { return } else { isInitialized = true }
-        let ptr = factories
+        let ptr = logHandlerFactories
```

Comment:
what is the purpose of assigning `logHandlerFactories` to a local?

p.s. what is assigned to the `ptr` local variable is not a pointer to a reference type as the name of the var implies.  It's a copy of `logHandlerFactories` which is a Swift Dictionary ( a value type.)